Blacklist command/inhibitor

UnseenFaith opened the pull request:
Just made a small inhibitor/command to deal with people who don't deserve to use commands/people I don't feel like dealing with on a moderation level so I just remove their access to commands completely lol. Might be useful for other people... could also extend this to per-server-confs, just a thought.

UnseenFaith committed:
blacklist.js Inhibitor

Adds blacklist inhibitor for checking the blacklist.json

UnseenFaith committed:
blacklist.js command

This command is used for blacklisting users from using commands. Used in conjunction with my blacklist inhibitor.

CyberiumShadow commented:
I think now that per-server confs are here. We can forgo the blacklist.json and integrate them into the guild confs. We just gotta wait for confs to fully work.

UnseenFaith commented:
Tis a good point, will work on this whenever I get per-server-confs working or when they work on install

UnseenFaith committed:
blacklist.js Inhibitor

Uses per-server-confs now

UnseenFaith committed:
blacklist.js command

Uses per-server-confs now!

UnseenFaith commented:
Updated PR with per-server-conf versions of both the inhibitor and command.

UnseenFaith committed:
Bugfix for Inhibitor

Forgot an issue where there Inhibitor could possibly return undefined on the blacklist if it's deleted from the JSON file while the bot is running... Fixed now

hkwu commented on commands/Moderation/blacklist.js:
```diff
@@ -0,0 +1,26 @@
+
+exports.run = (client, msg, cmd) => {
+    return new Promise((resolve, reject) => {
+      fs.readJson(`bwd/conf/${msg.guild.id}.json`, (err, guildConf) => {
```
You shouldn't need to use `fs` to read the confs. `client.funcs.confs` has a getter function to return a guild conf.

hkwu commented on inhibitors/blacklist.js:
```diff
@@ -0,0 +1,25 @@
+
+exports.run = (client, msg, cmd) => {
+    return new Promise((resolve, reject) => {
+      fs.readJson(`bwd/conf/${msg.guild.id}.json`, (err, guildConf) => {
```
Same as `commands/Moderation/blacklist.js`.

eslachance commented:
Thanks for the review @hkwu . Definitely correct, I have functions for that :D

UnseenFaith commented:
Yeah, I realized this last night. I'm going to update it to properly use the functions Evie made.. For whatever reason I didn't think that the functions would actually write to the file, hence why I did read/white.. nevertheless I'm going to fix it and commit it tonight.

CyberiumShadow commented:
@UnseenFaith Just letting you know that `<Message>.guildConf` can be used instead of needing to use `<Client>.funcs.confs.get(msg.guild)`

UnseenFaith committed:
Removed json read/write

UnseenFaith committed:
Removed Json Read/write

UnseenFaith committed:
Update 0.8.2

UnseenFaith committed:
Add files via upload

UnseenFaith commented:
Fixed the spaghetti code and updated both to use the latest Komada version (v.0.8.2 at the time of this writing). Hopefully the last commit that I have to do, but please do point out any more issues so I can fix them.

UnseenFaith committed:
Added blacklist.js in the wrong folder...

UnseenFaith committed:
Actual Final Commit (hopefully...)

CyberiumShadow commented:
@UnseenFaith Any updates to this now that your Komada PR has been merged?
